WordPress/Classes/ViewRelated/Reader/ReaderMenuViewModel.swift

```diff
@@ -0,0 +1,615 @@
+        sections.append(section)
+    }
+
+
+    /// Returns the number of items for the tags section.
+    ///
+    /// - Returns: The number of items in the section.
+    ///
+    func itemCountForTagSection() -> Int {
+        return tagsFetchedResultsController.fetchedObjects?.count ?? 0
+    }
+
+
+    /// Returns the menu item from the tags section at the specified index.
+    ///
+    /// - Parameters:
+    ///     - index: The index of the item.
+    ///
+    /// - Returns: The requested menu item or nil.
+    ///
+    func menuItemForTagAtIndex(index: Int) -> ReaderMenuItem? {
+        guard let topic = tagsFetchedResultsController.objectAtIndexPath(NSIndexPath(forRow: index, inSection: 0)) as? ReaderAbstractTopic else {
+            return nil
+        }
+        return ReaderMenuItem(title: topic.title, type: .Topic, icon: nil, topic: topic)
+    }
+
+
+    // MARK: - Helper Methods
+
+
+    /// Check if the user is signed in
+    ///
+    func isSignedInWPCom() -> Bool {
+        let service = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+        return service.defaultWordPressComAccount() != nil
+    }
+
+
+    /// Returns the predicate for tag and list fetch requests.
+    ///
+    /// - Returns: An NSPredicate
+    ///
+    func predicateForRequests() -> NSPredicate {
+        if isSignedInWPCom() {
+            return NSPredicate(format: "following = YES AND showInMenu = YES")
+        } else {
+            return NSPredicate(format: "following = NO AND showInMenu = YES")
+        }
+    }
+
+
+    // MARK: - Notifications
+
+
+    /// Handles a notification that the signed in wpcom account was changed.
+    /// All the content in the view model is updated as a result.
+    ///
+    func handleWordPressComAccountChanged(notification: NSNotification) {
+        // TODO: We need to ensure we have no stale topics in core data prior to reloading content.
+        delegate?.menuWillReloadContent()
+
+        // Update predicates to correctly fetch following or not following.
+        updateAndPerformListsFetchRequest()
+        updateAndPerformTagsFetchRequest()
+
+        setupSections()
+        delegate?.menuDidReloadContent()
+    }
+
+
+    // MARK: - View Model Interaction Methods
+
+
+    /// Returns the number of sections the menu should show.
+    ///
+    /// - Returns: The number of sections.
+    ///
+    func numberOfSectionsInMenu() -> Int {
+        return sections.count
+    }
+
+
+    /// Retuns the index of the section with the specified type.
+    ///
+    /// - Paramters:
+    ///     - type: The type of the section.
+    ///
+    /// - Return: The index of the section or -1 if it was not found.
+    ///
+    func indexOfSectionWithType(type: ReaderMenuSectionType) -> Int {
+        for (index, section) in sections.enumerate() {
+            if section.type == type {
+                return index
+            }
+        }
+        return -1
```

Comment:
I was wanting to avoid an optional but I suppose it would be more Swifty that way